### PR TITLE
Partial clean up of codebase

### PR DESCRIPTION
**Pull request by krishnakumarg1984**
Clean up incorporates (amid others) the following:
- removes duplicate header includes
- adds missing header includes
- adds closing namespace comments
- removes unnecessary semicolons (null statements)

### CODE REVIEW

File: cpp/examples/primal_dual/inpainting.cc [hunk 1]
```diff
@@ -16,7 +16,6 @@
 #include "sopt/power_method.h"
 #include "sopt/relative_variation.h"
 #include "sopt/sampling.h"
-#include "sopt/types.h"
```
Comment (tkoskela) [inline cpp/examples/primal_dual/inpainting.cc:19]: Why this removal? We are using `Vector`, `Matrix` and `Image` defined in `types.h` so I think we should explicitly include it, even though it is probably being included by some of the other headers.

**Review by tkoskela — CHANGES_REQUESTED** [review]
Cleanup looks good. I assume the standard library includes are all explicitly including things that are already being used in the code?

File: cpp/examples/primal_dual/inpainting.cc (hunk 1 above)
Comment (krishnakumarg1984) [reply cpp/examples/primal_dual/inpainting.cc:19]: @tkoskela  `types.h` was already included earlier in this file, and this second inclusion is a duplicate, which is being deleted in this PR.

**Comment by krishnakumarg1984** [comment]
> Cleanup looks good. I assume the standard library includes are all explicitly including things that are already being used in the code?

Yes.

File: cpp/examples/primal_dual/inpainting.cc (hunk 1 above)
Comment (tkoskela) [reply cpp/examples/primal_dual/inpainting.cc:19]: I see! Sorry I missed that

**Review by tkoskela — APPROVED** [review]

**Pull request merged**